spinalcordtoolbox/scripts/sct_deepseg.py

```diff
@@ -314,6 +320,25 @@ def main(argv: Sequence[str]):
         # Use the result of the current model as additional input of the next model
         fname_prior = fname_seg
 
+        # write JSON sidecar file
+        source_path = os.path.join(path_model, "source.json")
+        if os.path.isfile(source_path):
+            with open(source_path, "r") as fp:
+                source_dict = json.load(fp)
+        sidecar_json = {
+            'GeneratedBy': [
+                {
+                    "Name": f"spinalcordtoolbox: sct_deepseg {' '.join(os.path.basename(arg) for arg in argv)}",
+                    "Version": __version__,
+                    "CodeURL": f"https://github.com/spinalcordtoolbox/spinalcordtoolbox/"
+                               f"blob/{_git_info()[1].strip('*')}/spinalcordtoolbox/scripts/sct_deepseg.py",
+                    "ModelURL": source_dict["model_urls"][name_model][0],
```

Comment:
If you're ok with my other suggestions about `model_urls`, then it's also fine if this field is either-a-string-or-a-dict.